app/Main.hs

```diff
@@ -218,7 +219,9 @@ runMain options = do
     infos     = fmap (\ pc -> ProjectInfo (Config.owner pc) (Config.repository pc)) $ Config.projects config
   putStrLn $ "Listening for webhooks on port " ++ (show port) ++ "."
   runServer <- fmap fst $ buildServer port tlsConfig infos secret ghTryEnqueue getProjectState getOwnerState
-  runServer
+  serverThread <- Async.async runServer
 
   -- Note that a stop signal is never enqueued. The application just runs until
-  -- it is killed.
+  -- until any of the threads stop due to an exception (which may be due to a
+  -- sigterm, or due to an unhandled exception).
```

Comment:
Where do we handle SIGTERM? Without a signal handler, the default action for SIGTERM is killing the process, Haskell won't notice.

OTOH, SIGINT *is* converted into an exception, but it is thrown to the main thread, so to the one doing the `waitAny`. I'm not sure what happens in that case to the remaining threads if the main thread exits (are they stopped immediately or do they get a chance to clean up?).

Reply:
We don't have any signal handlers, will fix this comment.